utils/kernel.c

```diff
@@ -465,6 +465,46 @@ static int __setup_kernel_tracing(struct uftrace_kernel_writer *kernel)
 	return -EINVAL;
 }
 
+static void skip_kernel_functions(struct uftrace_kernel_writer *kernel)
+{
+	unsigned int i;
+	struct kfilter *kfilter;
+	const char *skip_funcs[] = {
+		"syscall_trace_enter_phase1",
+		"smp_irq_work_interrupt",
+		"syscall_slow_exit_work",
+	};
+
+	for (i = 0; i < ARRAY_SIZE(skip_funcs); i++) {
+		bool found = false;
+		const char *name = skip_funcs[i];
+		struct kfilter *pos, *tmp;
+
+		/* Don't skip it if user particularly want to see them*/
+		list_for_each_entry_safe(pos, tmp, &kernel->filters, list) {
+			if (strcmp(pos->name, name))
+				goto found;
+		}
+
+		list_for_each_entry_safe(pos, tmp, &kernel->patches, list) {
+			if (strcmp(pos->name, name)) {
+				found = true;
+				break;
+			}
+		}
+
+		if (found) {
+			found = false;
+found:
+			continue;
+		}
```

Comment:
This "found" logic looks not pretty.  What about this?

```
bool add = true;

list_for_each_entry(&kernel->filters)
  if (strcmp() == 0)
    add = false;
    break;

list_for_each_entry(*kernel->patches)
  if (strcmp() == 0)
    add = false;
    break;

if (add)
  list_add(&kernel->notrace);
```

Although it always traverses the 'filters' list and 'patches' list, I think it's fine as they're (usually) small lists.